extras/sdio/sdio.h

```diff
@@ -0,0 +1,228 @@
+        uint8_t c_size_mult_high   :2;
+        uint8_t vdd_w_cur_max      :3;
+        uint8_t vdd_w_curr_min     :3;
+        uint8_t sector_size_high   :6;
+        uint8_t erase_blk_en       :1;
+        uint8_t c_size_mult_low    :1;
+        uint8_t wp_grp_size        :7;
+        uint8_t sector_size_low    :1;
+        uint8_t write_bl_len_high  :2;
+        uint8_t r2w_factor         :3;
+        uint8_t reserved3          :2;
+        uint8_t wp_grp_enable      :1;
+        uint8_t reserved4          :5;
+        uint8_t write_partial      :1;
+        uint8_t write_bl_len_low   :2;
+        uint8_t reserved5          :2;
+        uint8_t file_format        :2;
+        uint8_t tmp_write_protect  :1;
+        uint8_t perm_write_protect :1;
+        uint8_t copy               :1;
+        uint8_t file_format_grp    :1;
+        uint8_t always1            :1;
+        uint8_t crc                :7;
+    } __attribute__((packed)) v1;
+
+    struct
+    {
+        uint8_t reserved1          :6;
+        uint8_t csd_ver            :2;
+        uint8_t taac;
+        uint8_t nsac;
+        uint8_t tran_speed;
+        uint8_t ccc_high;
+        uint8_t read_bl_len        :4;
+        uint8_t ccc_low            :4;
+        uint8_t reserved2          :4;
+        uint8_t dsr_imp            :1;
+        uint8_t read_blk_misalign  :1;
+        uint8_t write_blk_misalign :1;
+        uint8_t read_bl_partial    :1;
+        uint16_t c_size_high       :6;
+        uint8_t reserved3          :2;
+        uint8_t c_size_mid;
+        uint8_t c_size_low;
+        uint8_t sector_size_high   :6;
+        uint8_t erase_blk_en       :1;
+        uint8_t reserved4          :1;
+        uint8_t wp_grp_size        :7;
+        uint8_t sector_size_low    :1;
+        uint8_t write_bl_len_high  :2;
+        uint8_t r2w_factor         :3;
+        uint8_t reserved5          :2;
+        uint8_t wp_grp_enable      :1;
+        uint8_t reserved6          :5;
+        uint8_t write_partial      :1;
+        uint8_t write_bl_len_low   :2;
+        uint8_t reserved7          :2;
+        uint8_t file_format        :2;
+        uint8_t tmp_write_protect  :1;
+        uint8_t perm_write_protect :1;
+        uint8_t copy               :1;
+        uint8_t file_format_grp    :1;
+        uint8_t always1            :1;
+        uint8_t crc                :7;
+    } __attribute__((packed)) v2;
+} sdio_csd_t;
+
+/**
+ * SD card descriptor
+ */
+typedef struct
+{
+    sdio_error_t     error;        //!< Last operation result
+    uint8_t          cs_pin;       //!< Chip Select GPIO pin
+    sdio_card_type_t type;         //!< Card type
+    bool             crc_enabled;  //!< True if CRC enabled for IO
+    sdio_ocr_t       ocr;          //!< OCR register
+    sdio_csd_t       csd;          //!< CSD register
+    sdio_cid_t       cid;          //!< CID register
+    uint32_t         sectors;      //!< Card size in 512 byte sectors
+} sdio_card_t;
```

Comment:
All those structures are for internal usage. They are not intended to be manipulated from outside as far as I understand.
@UncleRus  do you think it is a good idea to hide them in some kind of sdio_private.h (or sdio_impl.h) and include it here? I understand it will not change the name space but it will reduce visual noise in the main sdio.h header? So, the main sdio.h will contain only dirver's interface. What do you think?